src/main/java/com/endava/endabank/controller/UserController.java

```diff
@@ -0,0 +1,89 @@
+package com.endava.endabank.controller;
+
+import com.endava.endabank.constants.Permissions;
+import com.endava.endabank.constants.Routes;
+import com.endava.endabank.dto.user.UpdatePasswordDto;
+import com.endava.endabank.dto.user.UserDetailsDto;
+import com.endava.endabank.dto.user.UserPrincipalSecurity;
+import com.endava.endabank.dto.user.UserRegisterDto;
+import com.endava.endabank.dto.user.UserToApproveAccountDto;
+import com.endava.endabank.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping(Routes.API_ROUTE + Routes.USERS_ROUTE)
+@AllArgsConstructor
+public class UserController {
+    private UserService userService;
+
+    @PostMapping
+    public Map<String, Object> create(@Valid @RequestBody UserRegisterDto user) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(userService.saveAndSendVerifyEmail(user)).getBody();
+    }
+
+    @GetMapping()
+    @PreAuthorize(Permissions.AUTHORITY_ACCOUNT_VALIDATE)
+    public ResponseEntity<List<UserToApproveAccountDto>> getUsersToApprove() {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.usersToApprove());
+    }
+
+    @PutMapping(Routes.APPROVE_ACCOUNT_ROUTE)
+    @PreAuthorize(Permissions.AUTHORITY_ACCOUNT_VALIDATE)
+    public ResponseEntity<UserToApproveAccountDto> updateUserIsApproved(@PathVariable Integer id, @RequestBody Map<String, Boolean> map) {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.updateUserAccountApprove(id, map.get("value")));
```

Comment:
|Hi,

What if somethings fail when "userAuthenticationService.logInUser(authentication)" is executed?

I think you should have a try-catch block to handle exceptions and throw custom exceptions instead of technical ones like BadDataException

if you put the returned entity always with ok you are gonna be handling always everything as ok

Please fix it in the same way for every controller class you have and its methods